src/metabase/query_processor/middleware/resolve.clj

```diff
@@ -230,11 +230,8 @@
 
   DateTimeField
   (parse-value [this value]
-    (let [tz                 (when-let [tz-id ^String (setting/get :report-timezone)]
-                               (TimeZone/getTimeZone tz-id))
-          parsed-string-date (some-> value
-                                     (u/str->date-time tz)
-                                     u/->Timestamp)]
+    (let [tz (or (setting/get :report-timezone) (System/getProperty "user.timezone"))
```

Comment:
This feels repeated from  https://github.com/metabase/metabase/pull/7520/files#diff-90aba221a9b8d7ad86ce77cad0535f91R601 and seems  like it should be abstracted out somewhere.